contracts/strategy/rysk/RyskStrategy.sol

```diff
@@ -0,0 +1,240 @@
+     */
+    constructor(
+        address _vault,
+        address _owner,
+        address _ryskLiquidityPool,
+        address _underlying
+    ) {
+        if (_owner == address(0)) revert StrategyOwnerCannotBe0Address();
+        if (_ryskLiquidityPool == address(0))
+            revert RyskLiquidityPoolCannotBe0Address();
+        if (_underlying == address(0))
+            revert StrategyUnderlyingCannotBe0Address();
+
+        if (!_vault.doesContractImplementInterface(type(IVault).interfaceId))
+            revert StrategyNotIVault();
+
+        _setupRole(DEFAULT_ADMIN_ROLE, _owner);
+        _setupRole(MANAGER_ROLE, _vault);
+
+        vault = _vault;
+        ryskLqPool = IRyskLiquidityPool(_ryskLiquidityPool);
+        underlying = IERC20(_underlying);
+
+        underlying.approve(_ryskLiquidityPool, type(uint256).max);
+    }
+
+    //
+    // Modifiers
+    //
+
+    modifier onlyManager() {
+        if (!hasRole(MANAGER_ROLE, msg.sender))
+            revert StrategyCallerNotManager();
+        _;
+    }
+
+    //
+    // Ownable
+    //
+
+    /**
+     * Transfers ownership of the Strategy to another account,
+     * revoking previous owner's ADMIN role and setting up ADMIN role for the new owner.
+     *
+     * @notice Can only be called by the current owner.
+     *
+     * @param _newOwner The new owner of the contract.
+     */
+    function transferOwnership(address _newOwner)
+        public
+        override(Ownable)
+        onlyOwner
+    {
+        if (_newOwner == address(0x0)) revert StrategyOwnerCannotBe0Address();
+        if (_newOwner == msg.sender)
+            revert StrategyCannotTransferOwnershipToSelf();
+
+        _transferOwnership(_newOwner);
+
+        _setupRole(DEFAULT_ADMIN_ROLE, _newOwner);
+
+        _revokeRole(DEFAULT_ADMIN_ROLE, msg.sender);
+    }
+
+    //
+    // IStrategy
+    //
+
+    /**
+     * Rysk strategy is asynchronous meaning it doesn't support immediate withdrawals.
+     *
+     * @return false always
+     */
+    function isSync() external pure override(IStrategy) returns (bool) {
+        return false;
+    }
+
+    /// @inheritdoc IStrategy
+    function hasAssets()
+        external
+        view
+        virtual
+        override(IStrategy)
+        returns (bool)
+    {
+        return _getTotalShares() != 0;
+    }
+
+    /// @inheritdoc IStrategy
+    function investedAssets()
+        external
+        view
+        virtual
+        override(IStrategy)
+        returns (uint256)
+    {
+        return _sharesToUnderlying(_getTotalShares());
+    }
+
+    /// @inheritdoc IStrategy
+    function invest() external virtual override(IStrategy) onlyManager {
+        uint256 balance = underlying.balanceOf(address(this));
+
+        if (balance == 0) revert StrategyNoUnderlying();
+
+        ryskLqPool.deposit(balance);
+
+        emit StrategyInvested(balance);
+    }
+
+    /// @inheritdoc IStrategy
+    function withdrawToVault(uint256 amount)
```

Comment:
I wonder if we shouldn't be tracking the ongoing withdrawals in case the vault rebalances multiple times before the withdrawal is completed

Reply:
good point :+1: I will think about how to implement this since we really don't have a way to know how long an epoch can last